src/main/java/com/redhat/devtools/intellij/tektoncd/actions/component/ShowLogsAction.java

```diff
@@ -0,0 +1,106 @@
+import com.redhat.devtools.intellij.tektoncd.tree.PipelineNode;
+import com.redhat.devtools.intellij.tektoncd.tree.PipelineRunNode;
+import com.redhat.devtools.intellij.tektoncd.tree.TaskNode;
+import com.redhat.devtools.intellij.tektoncd.tree.TaskRunNode;
+import com.redhat.devtools.intellij.tektoncd.ui.component.RunPickerDialog;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.tree.TreePath;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.redhat.devtools.intellij.tektoncd.Constants.NOTIFICATION_ID;
+
+public class ShowLogsAction extends TektonAction {
+    Logger logger = LoggerFactory.getLogger(ShowLogsAction.class);
+
+    public ShowLogsAction() { super(PipelineRunNode.class, TaskRunNode.class, TaskNode.class, PipelineNode.class); }
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Tkn tkncli) {
+        String namespace;
+        String resourceName = selected.toString();
+        List<String> resourceRunsName = null;
+        Class<?> nodeClass = selected.getClass();
+        String kindLabel = "";
+        Notification notification;
+        // if node selected is a Pipeline or a Task we need to find its runs
+        try {
+            namespace = ((LazyMutableTreeNode)selected).getParent().getParent().toString();
+            if (PipelineNode.class.equals(nodeClass)) {
+                List<PipelineRun> pipelineRuns = tkncli.getPipelineRuns(namespace, resourceName);
+                if (pipelineRuns == null || pipelineRuns.size() == 0) {
+                    throw new IOException("Pipeline " + resourceName + "doesn't have any pipelineRun to be selected");
+                }
+                resourceRunsName = pipelineRuns.stream().map(PipelineRun::getName).collect(Collectors.toList());
+                nodeClass = PipelineRunNode.class;
+                kindLabel = "pipelinerun";
+            } else if (TaskNode.class.equals(nodeClass)) {
+                List<TaskRun> taskRuns = tkncli.getTaskRuns(namespace, resourceName);
+                if (taskRuns == null || taskRuns.size() == 0) {
+                    throw new IOException("Task " + resourceName + "doesn't have any taskRun to be selected");
+                }
+                resourceRunsName = taskRuns.stream().map(TaskRun::getName).collect(Collectors.toList());
+                nodeClass = TaskRunNode.class;
+                kindLabel = "taskrun";
+            }
+        } catch (IOException e) {
+            notification = new Notification(NOTIFICATION_ID, "Error", "An error occurred while requesting logs for " + resourceName + "\n" + e.getLocalizedMessage(), NotificationType.ERROR);
+            Notifications.Bus.notify(notification);
+            logger.error("Error: " + e.getLocalizedMessage(), e);
+            return;
+        }
+
+        if (resourceRunsName != null) {
+            if (resourceRunsName.size() == 1) {
```

Comment:
We need to report an message to user if resourceRunsName is empty (Messages)

Reply:
The error is shown inside the try catch above. Isn't that enough? The pipeline/task run lists are checked and if found empty a message is thrown e.g `Pipeline blabla doesn't have any pipelineRun to be selected`

Reply:
I did not see the test but I don't think we should report the same way as an error:
- if the user did something wrong I think the dialog should be an information one and not an error one
- don't forget that error dialogs are quite vague and the information may not be quite clear to the user. See https://github.com/redhat-developer/intellij-openshift-connector/blob/master/src/main/java/org/jboss/tools/intellij/openshift/actions/component/LinkServiceAction.java as an example